src/TraceEvent/Parsers/ClrTraceEventParser.cs

```diff
@@ -4941,8 +4941,8 @@ public enum Gens
         Gen1,
         Gen2,
         GenLargeObj,
-        Gen0After,
         GenPinObj,
+        GenCount, 
```

Comment:
we do need to consider backward compatibility here - in 3.1 and priori there were 4 elements in this array instead of 5, we still need it to work with those older versions.

Reply:
I am wondering about the backward compatibility requirements here. 

If we keep `Gen0After` to be 4, and somehow the value is used as the length to construct an array to put things in, and then the trace contains a POH, then the code will end up having an `ArrayOutOfBoundsException`. This change will probably help in the sense that when the user upgrades the TraceEvent package, they will get a compile error, and notice there is a need to change things for POH.

The ideal case is probably to avoid this problem in the first place by not having such an enum entry at all.